pkg/minikube/registry/drvs/docker/docker.go

```diff
@@ -86,57 +87,88 @@ func status() registry.State {
 	ctx, cancel := context.WithTimeout(context.Background(), 6*time.Second)
 	defer cancel()
 
-	// Quickly returns an error code if server is not running
 	cmd := exec.CommandContext(ctx, oci.Docker, "version", "--format", "{{.Server.Os}}-{{.Server.Version}}")
 	o, err := cmd.Output()
-	output := string(o)
-	if strings.Contains(output, "windows-") {
-		return registry.State{Error: oci.ErrWindowsContainers, Installed: true, Healthy: false, Fix: "Change container type to \"linux\" in Docker Desktop settings", Doc: docURL + "#verify-docker-container-type-is-linux"}
+	if err != nil {
+		if ctx.Err() == context.DeadlineExceeded {
+			err = errors.Wrapf(err, "deadline exceeded running %q", strings.Join(cmd.Args, " "))
+		}
+
+		glog.Warningf("docker version returned error: %v", err)
+
+		if exitErr, ok := err.(*exec.ExitError); ok {
+			stderr := strings.TrimSpace(string(exitErr.Stderr))
+			newErr := fmt.Errorf(`%q %v: %s`, strings.Join(cmd.Args, " "), exitErr, stderr)
+
+			return suggestFix(stderr, newErr)
+		}
 
+		return registry.State{Error: err, Installed: true, Healthy: false, Fix: "Restart the Docker service", Doc: docURL}
 	}
-	if err == nil {
-		glog.Infof("docker version: %s", output)
-		return checkNeedsImprovement()
+
+	glog.Infof("docker version: %s", o)
+	if strings.Contains(string(o), "windows-") {
+		return registry.State{Error: oci.ErrWindowsContainers, Installed: true, Healthy: false, Fix: "Change container type to \"linux\" in Docker Desktop settings", Doc: docURL + "#verify-docker-container-type-is-linux"}
 	}
 
-	glog.Warningf("docker returned error: %v", err)
+	ctx, cancel = context.WithTimeout(context.Background(), 6*time.Second)
+	defer cancel()
+
+	// Why run "info"? On some releases (Docker Desktop v19.03.12), "version" returns exit code 0 even if Docker is down.
+	cmd = exec.CommandContext(ctx, oci.Docker, "info")
```

Comment:
please use CachedDaemonInfo 

https://github.com/medyagh/minikube/blob/fd44e0d8fb93e317d4e8c40558e8a5120ea17206/pkg/drivers/kic/oci/info.go#L41

Reply:
Great idea. I had to plumb errors through via JSON, which was cleaner than my approach. The PR output changes in a minor way, so I've updated that.